### ISSUE TITLE
Unexpected behavior of min.code.count 

### ISSUE THREAD

**Issue by amlake** [PheWAS/PheWAS#58]
I am noticing that the min.code.count parameter in `createPhewasTable` and` createPhenotypes` is not working as I had expected. I thought that setting `min.code.count=2`, for example, would require at least 2 instances of any ICD code falling under the parent phecode in order to classify an individual as a case. However, upon further investigation, in my dataset, it appears that in order be classified as a case, an individual must have at least 2 instances of **any one ICD code** falling under the parent phecode. For example, if an individual has one instance of the ICD-9 code `295.00` and one instance of the ICD-10 code `F20.2`, despite having 2 total instances of ICD codes falling under the parent phecode `295.1` (Schizophrenia), they will still be counted as only having 1 instance of `295.1`  and therefore be excluded from the analysis if `min.code.count` is set to 2.  I have only confirmed this in a few example cases, and I have not yet assessed this on a systematic level in my dataset, but I wanted to check with the package creators in the meantime. Thanks for your help!

**Comment by amlake** [comment]
It seems that the issue is with the `make.distinct` parameter. It looks like if the input icd code count data frame does **not** have a date column, the `make.distinct` parameter being set by default to `TRUE` has unintended consequences.

In this example, individual **R1** has 3 instances of ICD codes falling under phecodes `295` and `295.1`. However, since  `make.distinct=TRUE` causes repeated rows to be filtered out after the conversion from ICD codes to phecodes, this individual will be only counted as having **1** instance of phecode`295` and **1** instance of phecode`295.1` (instead of 3 each). Therefore they will be misclassified as "NA" instead of as a case for those phenotypes.

```
> # create some fake ICD count data with no date column
> dat <- data.frame(grid = c("R1","R1","R1","R2","R2"),
+                   code = c("295.0","295.2","F20.0","295.0","295.55"),
+                   index = c(1,1,1,2,1),
+                   vocabulary_id = c("ICD9CM","ICD9CM","ICD10CM","ICD9CM","ICD9CM"))
> 
> dat_mapped <- mapCodesToPhecodes(dat)
> dat_mapped
  grid index phecode
1   R1     1   295.1
2   R1     1     295
3   R2     2   295.1
4   R2     2     295
5   R2     1   295.1
6   R2     1     295

> phecode <- createPhewasTable(dat_mapped[,c(1,3,2)], min.code.count=2, add.exclusions=T, translate=F)
> as.data.frame(phecode[,1:3])
  grid  295 295.1
1   R1   NA    NA
2   R2 TRUE  TRUE
```

In this example, setting `make.distinct=FALSE` fixes this behavior:
```
> dat_mapped2 <- mapCodesToPhecodes(dat, make.distinct = F)
> dat_mapped
  grid index phecode
1   R1     1   295.1
2   R1     1     295
3   R2     2   295.1
4   R2     2     295
5   R2     1   295.1
6   R2     1     295

> phecode2 <- createPhewasTable(dat_mapped2[,c(1,3,2)], min.code.count=2, add.exclusions=T, translate=F)
> as.data.frame(phecode2[,1:3])
  grid  295 295.1
1   R1 TRUE  TRUE
2   R2 TRUE  TRUE
```

In this example, you can see that `make.distinct=TRUE` works as expected if the input data frame includes a date column:

```
> dat2 <- cbind(dat, data.frame(date=c("date1","date1","date2","date1","date3")))
> dat_mapped3 <- mapCodesToPhecodes(dat2, make.distinct = T)
> dat_mapped3
  grid index  date phecode
1   R1     1 date1   295.1
2   R1     1 date1     295
3   R1     1 date2   295.1
4   R1     1 date2     295
5   R2     2 date1   295.1
6   R2     2 date1     295
7   R2     1 date3   295.1
8   R2     1 date3     295
> 
> phecode3 <- createPhewasTable(dat_mapped3[,c(1,4,2)], min.code.count=2, add.exclusions=T, translate=F)
> as.data.frame(phecode3[,1:3])
  grid  295 295.1
1   R1 TRUE  TRUE
2   R2 TRUE  TRUE
```

**Comment by RobertJCarroll** [comment]
Thanks for pointing this out. I will address this.

**Comment by RobertJCarroll** [comment]
I don't think this affects `createPhewasTable` does it? I know the example does

On further investigation, I understand the differences between expectations and what is observed. This default behavior helps somewhat in an era of dual-coding, but it's reasonable to update now with a different default behavior.

**Comment by amlake** [comment]
I don't think it directly affects `createPhewasTable`, no. But it does affect `createPhenotypes`, which uses `mapCodesToPhecodes`. Thanks for looking into this!

**Comment by RobertJCarroll** [comment]
Thank you for confirming. 

I have a strategy in mind that accounts for communicating the potentially changing behavior, but it will take a little time to implement.

**Comment by RobertJCarroll** [comment]
Apologies for the delay- I made these changes. I was worried about potential cascading effects, but I believe they are handled sufficiently.
